lisa/tools/cargo.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import PurePath
+from typing import Optional, cast
+from typing import List, Type
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+from lisa.tools.echo import Echo
+from lisa.util.process import ExecutableResult
+
+from .gcc import Gcc
+
+
+class Cargo(Tool):
+
+    @property
+    def command(self) -> str:
+        return "cargo"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Gcc]
+
+    def _install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        cargo_source_url = "https://sh.rustup.rs"
+        try:
+            # install prerequisites
+            posix_os.install_packages(["build-essential", "curl", "cmake"])
+            gcc = self.node.tools[Gcc]
+            gcc_version_info = gcc.get_version().to_dict()
+            self.node.log.info(f"Gcc Version: {gcc_version_info}")
+
```

Comment:
I didn't see you used these in install method

Reply:
This is just to print Gcc and Curl version (newly added) so we can have it in log while installing cargo. This will help in case of cargo installation failed.